utilities/cli/holohub.py

```diff
@@ -349,12 +365,33 @@ def handle_build_container(self, args: argparse.Namespace) -> None:
             project_name=args.project, language=args.language if hasattr(args, "language") else None
         )
         container.dryrun = args.dryrun
+
+        # Get mode-specific build arguments if project is specified
+        mode_build_args = ""
+        if args.project:
+            project_data = self._find_project(
+                project_name=args.project,
+                language=args.language if hasattr(args, "language") else None,
+            )
+            mode_config = self._get_mode_config(project_data, args.mode)
+            build_config = mode_config.get("build", {})
+
+            # Apply mode-specific docker build args
+            build_docker_args = build_config.get("docker_args", [])
+            if isinstance(build_docker_args, list):
+                mode_build_args = " ".join(build_docker_args)
+
+        # Combine user-provided build args with mode-specific args
+        combined_build_args = args.build_args or ""
```

Comment:
Nitpick, should we set build_args with `default=''` instead of trying to handle the `None` default?